samples/news/News.UITests/Tests.cs

```diff
@@ -62,6 +62,31 @@ public void SetLightMode()
                 .SetLightMode();
         }
 
+        [Test]
+        public void SwipeThroughEachCategory()
+        {
+            var homePage = new HomePage();
```

Comment:
I don't really like this syntax and style of test writing, it makes no sense to me what is happening. I want the tests to be following the pattern we have and the idea being they are easily readable and give a sense of what is being tested. This is the syntax pattern our tests should follow : 

```
 new HomePage()
         .ShowSearchTab()
         .Search("Microsoft");
```
Also loops inside a test is asking for trouble ,if one category fails, the rest of the test will fail. Small and independent tests is what I think is the best approach. 

My suggestion would be to use NUnit [Category] tags and remove the foreach loop. Divide this into separate tests. 

Why independent tests are important (Also referenced in my blog post) :
https://github.com/xamarin-automation-service/uitest-pop-example/wiki/General-Testing-Suggestions#split-tests-up-into-small-chunks-wherever-possible 

NUnit Category Tag : http://nunit.org/docs/2.2/category.html

Reply:
that is exactly the pattern we used to have so I was following the existing test:

<img width="1893" alt="commit-uitests" src="https://user-images.githubusercontent.com/3697084/54642232-575c8b80-4a51-11e9-9656-ee9aaf09906d.png">

https://github.com/xamarin/mobcat/commit/d5e273d61f6471c6eb8ff74e2b99c6c91923b178#diff-2e95bafb133ac0253594a183cb8dae8a

Reply:
I've reimplemented test and created a new instance of a page object whenever it's out of execution chain. Please check the updated source

Reply:
> Also loops inside a test is asking for trouble, if one category fails, the rest of the test will fail. Small and independent tests are what I think is the best approach.

this test is testing not a swipe action but sequence of categories and sources. Basically, this is the smallest chunk possible to test that all categories and sources are present in a given number and order. Is it possible to implement it with the `NUnit Category Tag`?